molecule/default/tasks/template.yml

```diff
@@ -0,0 +1,154 @@
+      community.kubernetes.k8s_service:
+        template: "../templates/pod_template_one.j2"
+        state: present
+      ignore_errors: yes
+      register: r
+
+    - name: Check for expected failures in last tasks
+      assert:
+        that:
+          - r.failed
+          - "'is only supported parameter for' in r.msg"
+
+    - name: Specify both definition and template
+      community.kubernetes.k8s:
+        state: present
+        template: "../templates/pod_template_one.j2"
+        definition:
+          apiVersion: apps/v1
+          kind: Deployment
+          metadata:
+            name: apply-deploy
+            namespace: "{{ template_namespace }}"
+          spec:
+            replicas: 1
+            selector:
+              matchLabels:
+                app: "{{ k8s_pod_name }}"
+      vars:
+        k8s_pod_name: pod
+        k8s_pod_namespace: "{{ template_namespace }}"
+      register: r
+      ignore_errors: yes
+
+    - name: Check if definition and template are mutually exclusive
+      assert:
+        that:
+          - r.failed
+          - "'parameters are mutually exclusive' in r.msg"
+
+    - name: Specify both src and template
+      community.kubernetes.k8s:
+        state: present
+        src: "../templates/pod_template_one.j2"
+        template: "../templates/pod_template_one.j2"
+      vars:
+        k8s_pod_name: pod
+        k8s_pod_namespace: "{{ template_namespace }}"
+      register: r
+      ignore_errors: yes
+
+    - name: Check if src and template are mutually exclusive
+      assert:
+        that:
+          - r.failed
+          - "'parameters are mutually exclusive' in r.msg"
+
+    - name: Create pod using template
+      community.kubernetes.k8s:
+        template: "../templates/pod_template_one.j2"
+        wait: yes
+      vars:
+        k8s_pod_name: pod-1
+        k8s_pod_namespace: "{{ template_namespace }}"
+      register: r
+
+    - name: Assert that pod creation succeeded using template
+      assert:
+        that:
+          - r is successful
+
+    - name: Create pod using template
+      community.kubernetes.k8s:
+        template:
+          - default
+        wait: yes
+      vars:
+        k8s_pod_name: pod-2
+        k8s_pod_namespace: "{{ template_namespace }}"
+      register: r
+      ignore_errors: True
+
+    - name: Assert that pod creation failed using template
+      assert:
+        that:
+          - r is failed
+          - "'Error while reading template file' in r.msg"
+
+    - name: Create pod using template
+      community.kubernetes.k8s:
+        template:
+          path: "../templates/pod_template_one.j2"
+        wait: yes
+      vars:
+        k8s_pod_name: pod-3
+        k8s_pod_namespace: "{{ template_namespace }}"
+      register: r
+
+    - name: Assert that pod creation succeeded using template
+      assert:
+        that:
+          - r is successful
+
+    - name: Create pod using template
```

Comment:
Can you change desc here to "Create pod using template (different variable string)"